app/src/main/java/com/example/amcc/view/MainFragment.java

```diff
@@ -201,10 +203,10 @@ private boolean inputIsValid() {
 
         validation.addValidation(avgConField, "(^[1-9]\\d?(\\.[0-9]\\d?)?$)", errorMsg);
         View error = city;
-        if (!cityFromApi.contains(city.getText().toString())) {
-
+        String location = city.getText().toString();
+        if (!cityFromApi.contains(location) && !viewModel.mPostCodes.contains(location)) {
```

Comment:
Why is the if statment && und not || ?

Reply:
if it is not a city and it is not a postcode then show error